pkg/controller/controller_test.go

```diff
@@ -383,10 +394,106 @@ func TestRolloutController_Reconcile(t *testing.T) {
 			kubePatchErr:        errors.New("cannot patch StatefulSet right now"),
 			expectedDeletedPods: []string{"ingester-zone-b-0", "ingester-zone-b-1"},
 		},
+		"should return early and scale up statefulset based on target custom resource": {
```

Comment:
[nit] To keep naming consistency

```suggestion
		"should return early and scale up statefulset based on reference custom resource": {
```